src/sentry/integrations/slack/views/unlink_identity.py

```diff
@@ -35,37 +42,91 @@ class SlackUnlinkIdentityView(BaseView):
-        except (SignatureExpired, BadSignature):
+        except (SignatureExpired, BadSignature) as e:
+            logger.warning("dispatch.signature_error", exc_info=e)
+            metrics.incr(self._METRICS_FAILURE_KEY, tags={"error": str(e)}, sample_rate=1.0)
             return render_to_response(
                 "sentry/integrations/slack/expired-link.html",
                 request=request,
             )
+        return super().dispatch(request, params=params)
+
+    def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        params = kwargs["params"]
+        method = ".get"
+
+        try:
+            organization, integration, idp = get_identity_or_404(
+                ExternalProviders.SLACK,
+                request.user,
+                integration_id=params["integration_id"],
+            )
+            logger.info("get_identity_success", extra={"integration_id": params["integration_id"]})
+            metrics.incr(
+                self._METRICS_SUCCESS_KEY + method, tags={"action": "get_identity"}, sample_rate=1.0
+            )
+        except Http404:
+            logger.exception(
+                "get_identity_error", extra={"integration_id": params["integration_id"]}
+            )
+            metrics.incr(
+                self._METRICS_FAILURE_KEY + method, tags={"action": "get_identity"}, sample_rate=1.0
+            )
+            raise
 
-        organization, integration, idp = get_identity_or_404(
-            ExternalProviders.SLACK,
-            request.user,
-            integration_id=params["integration_id"],
+        return render_to_response(
+            "sentry/auth-unlink-identity.html",
+            request=request,
+            context={"organization": organization, "provider": integration.get_provider()},
         )
 
-        if request.method != "POST":
-            return render_to_response(
-                "sentry/auth-unlink-identity.html",
-                request=request,
-                context={"organization": organization, "provider": integration.get_provider()},
+    def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        method = ".post"
+        params = kwargs["params"]
+        try:
+            organization, integration, idp = get_identity_or_404(
+                ExternalProviders.SLACK,
+                request.user,
+                integration_id=params["integration_id"],
+            )
+            logger.info("get_identity_success", extra={"integration_id": params["integration_id"]})
+            metrics.incr(
+                self._METRICS_SUCCESS_KEY + method, tags={"action": "get_identity"}, sample_rate=1.0
             )
+        except Http404:
+            logger.exception(
+                "get_identity_error", extra={"integration_id": params["integration_id"]}
+            )
+            metrics.incr(
+                self._METRICS_FAILURE_KEY + method, tags={"action": "get_identity"}, sample_rate=1.0
```

Comment:
should the metrics here be `post_identity`

Reply:
i moved the function call to `dispatch` so we don't repeat it